transport/memory.go

```diff
@@ -0,0 +1,60 @@
+package transport
+
+import (
+	"context"
+
+	"github.com/spacemeshos/poet/registration"
+	"github.com/spacemeshos/poet/service"
+	"github.com/spacemeshos/poet/shared"
+)
+
+type transport interface {
+	service.RegistrationService
+	registration.WorkerService
+}
+
+// inMemory is an in-memory implementation of transport.
+// It allows binding a Registration service with a Worker service
+// in a standalone mode by an in-memory channel.
+type inMemory struct {
+	closedRounds chan service.ClosedRound
+	proofs       chan shared.NIP
+}
+
+func NewInMemory() transport {
```

Comment:
I'm not sure, it seems an overkill at the moment. It would make the implementation more complex than it needs to be. I'd reconsider it when additional transport is actually required (when we do the split into separate processes in #365. Wdyt?

Reply:
In the long run I think it would make it simpler than complex. It would force us to clearly define our DTOs and service internal data structures which leads to better decoupling and easier testing: it would also avoid the cyclic dependency issue between `config`, `round_config` and other packages. If they are defined in `server` they can only be used there and other services would need their own configuration (that might be passed via an API or loaded during startup, depending on the use case).

Yes, within this PR this would be too much, but I feel like it is worth giving a try for #365